server/src/test/java/org/apache/kafka/server/share/CachedSharePartitionTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.share;
+
+import org.apache.kafka.common.Uuid;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+public class CachedSharePartitionTest {
+    @Test
+    public void testCachedSharePartitionEqualsAndHashCode() {
+        Uuid topicId = Uuid.randomUuid();
+        String topicName = "topic";
+        int partition = 0;
+
+        CachedSharePartition cachedSharePartitionWithIdAndName = new
+                CachedSharePartition(topicName, topicId, partition, false);
+        CachedSharePartition cachedSharePartitionWithIdAndNoName = new
+                CachedSharePartition(null, topicId, partition, false);
```

Comment:
are we sure about this, because I see test `testCachedPartitionEqualsAndHashCode` in FetchSessionTest.scala where topic name is null. [Code reference](https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/server/FetchSessionTest.scala#L1881)

Reply:
Seems very strange why it should be null. I checked `TopicPartition` class which also don't have it as null so may be we can leave it like this only.